### PR TITLE
Common block in /info API route

### PR DESCRIPTION
**Pull request by catena2w**
Variant 1 fix of https://github.com/ergoplatform/ergo/issues/328

### CODE REVIEW

File: src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/HeadersProcessor.scala [hunk 1]
```diff
@@ -66,6 +66,20 @@ trait HeadersProcessor extends ToDownloadProcessor with ScorexLogging with Score
     */
   def headersHeight: Int = bestHeaderIdOpt.flatMap(id => heightOf(id)).getOrElse(-1)
 
+  /**
+    * Common block between bestHeader and bestFullBlock
+    */
+  def commonBlockIdOpt: Option[ModifierId] = {
+    bestFullBlockOpt match {
```
Comment (mike-aksarin) [inline src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/HeadersProcessor.scala]: maybe `bestFullBlockOpt flatMap { fb =>` will look better here?

File: src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/HeadersProcessor.scala [hunk 2]
```diff
@@ -66,6 +66,18 @@ trait HeadersProcessor extends ToDownloadProcessor with ScorexLogging with Score
     */
   def headersHeight: Int = bestHeaderIdOpt.flatMap(id => heightOf(id)).getOrElse(-1)
 
+  /**
+    * Common block between bestHeader and bestFullBlock
+    */
+  def commonBlockIdOpt: Option[ModifierId] = {
+    bestFullBlockOpt.flatMap { fb =>
+        headerChainBack(config.blocksToKeep,
```
Comment (kushti) [inline src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/HeadersProcessor.scala]: Why this limit? A common header could be deeper than that.
An independent question is what could be lowest limit on config.blocksToKeep (I propose to set it to be config.keepVersions at least).

Comment (kushti) [reply src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/HeadersProcessor.scala]: commonBlockIdOpt implementation is utterly inefficient. For every call a lot of disk reads will be done potentially. Isn't it better to store and update a variable storing the reference to the common header?

**Comment by catena2w** [comment]
should be useless after https://github.com/ergoplatform/ergo/pull/333 merge

**Pull request closed without merging**